Add a numeric value label above each bar

Jerry-Sun opened the pull request:
Adding value labels above each bar in the graph to provide viewer with additional detail

Jerry-Sun committed:
Bar Graph Value Labels

Adding value labels above each bar in the graph to provide viewer with additional detail

Jerry-Sun committed:
Update main.py

dylansteele8 commented on main.py:
```diff
@@ -53,13 +53,17 @@ def analyze_messages(inbox_path, number_conversations):
     )
+    for bar in bars:
+        yval = bar.get_height()
+        plt.text(bar.get_x() + bar.get_width()/2, yval * 1.015, yval, ha='center', size=8)
```
I think it makes more sense to have a constant y-offset for the height of the label. Something like `yval + 10` instead of `yval * 1.015`.

dylansteele8 replied:
Would it be possible to get the label to format with thousands separators?

Jerry-Sun replied:
I tried to use a constant y-offset but it wasn't working regardless if I used yval + 1, yval + 10, or yval + 100.

dylansteele8 replied:
What do you by it wasn't working? What are the units of the height argument? I thought it would be pixels, but thinking about it some more it's probably the actual y-value. So, for example, you're looking at `yval = 100,000` so adding 100 isn't going to do much. I would probably add 2(ish)% of the max y-value.

Jerry-Sun replied:
I tried to use a constant y-offset but it wasn't working regardless if I used ```yval + 1```, ```yval + 10```, or ```yval + 100```.

dylansteele8 replied:
I think something like this should work and would allow for a constant offset for all labels.

```python
    y_offset = max(nlargest_messages[1]) * 0.015
    for bar in bars:
        yval = bar.get_height()
        plt.text(
            bar.get_x() + bar.get_width() / 2,
            yval + y_offset,
            yval,
            ha="center",
            size=8,
        )
```

dylansteele8 replied:
And changing the third argument from `yval` to `f"{yval:,}"` will add thousands separators.

Jerry-Sun replied:
> I think something like this should work and would allow for a constant offset for all labels.
> 
> ```python
>     y_offset = max(nlargest_messages[1]) * 0.015
>     for bar in bars:
>         yval = bar.get_height()
>         plt.text(
>             bar.get_x() + bar.get_width() / 2,
>             yval + y_offset,
>             yval,
>             ha="center",
>             size=8,
>         )
> ```

Oh, I understand what you are saying. Good solution.

dylansteele8 commented on main.py:
```diff
@@ -53,13 +53,17 @@ def analyze_messages(inbox_path, number_conversations):
+    for bar in bars:
+        yval = bar.get_height()
+        plt.text(bar.get_x() + bar.get_width()/2, yval * 1.015, yval, ha='center', size=8)
+    plt.tight_layout()
```
What does this do?

Jerry-Sun replied:
Adds the labels to the bars. Goes through and adds the text in the appropriate position.

Jerry-Sun replied:
Adds the labels to the bars. ```yval = bar.get_height()``` extracts the value while ```plt.text(...)``` places the label in the correct location above each bar.

Jerry-Sun committed:
Update main.py

Jerry-Sun committed:
Update main.py

Make positioning of value labels constant above each bar and add the thousands comma

Jerry-Sun committed:
Update main.py

Removed the ```import pandas as pd``` at the top. That was used to explore the racing bar graph feature.
